### PR TITLE
Prototype for reworking the Scene class.

### PR DESCRIPTION
**Pull request by moorepants**
@sahilshekhawat This is my rework of the scene class that should allow you to complete the PR you were working on. Can you pull the branch, add the addtional things you were working on in #196 on top of these commits and send in a new PR?

### CODE REVIEW

**Comment by moorepants** [comment]
@sahilshekhawat Do you want to handle this further or shall I?

**Comment by sahilshekhawat** [comment]
Should I add more commits here consists of what I have already done on #196 and more?

**Comment by sahilshekhawat** [comment]
I am gonna take your changes as is and raise another PR. This is the last issue blocking the release. I want to fix it as soon as possible.

**Comment by moorepants** [comment]
Please preserve commit history if possible.

**Comment by sahilshekhawat** [comment]
Yes, it will be. I have fetched this PR and will push that branch itself.

File: pydy/viz/scene.py [hunk 1]
```diff
@@ -154,6 +175,146 @@ def reference_frame(self, new_reference_frame):
...
             self._reference_frame = new_reference_frame
 
+    @property
+    def system(self):
+        return self._system
+
+    @system.setter
+    def system(self, new_system):
+
+        if new_system is not None and not isinstance(new_system, System):
```
Comment (oliverlee) [inline pydy/viz/scene.py]: You don't need to explicitly check if it is None.

Comment (sahilshekhawat) [inline pydy/viz/scene.py]: Can you please review #242 instead of this , I have added some extra things to this PR.

**Comment by moorepants** [comment]
I'll close this one. Thanks for using it.

**Pull request closed without merging**